travel/src/main/java/com/tx/travel/controller/DailyAllowanceController.java

```diff
@@ -0,0 +1,97 @@
+package com.tx.travel.controller;
+
+import com.tx.travel.payload.request.DailyAllowanceRequest;
+import com.tx.travel.payload.response.DailyAllowanceResponse;
+import com.tx.travel.payload.response.MessageResponse;
+import com.tx.travel.service.DailyAllowanceService;
+import com.tx.travel.service.exception.DailyAllowanceNotFoundException;
+import com.tx.travel.service.exception.RegionAlreadyExistsException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@CrossOrigin(origins = "*", maxAge = 3600)
+@RestController
+@RequestMapping("/api/daily-allowances")
+public class DailyAllowanceController {
+
+    final DailyAllowanceService dailyAllowanceService;
+
+    public DailyAllowanceController(final DailyAllowanceService dailyAllowanceService) {
+        this.dailyAllowanceService = dailyAllowanceService;
+    }
+
+    @GetMapping
+    //@PreAuthorize("hasRole('ROLE_EMPLOYEE') or hasRole('ROLE_OFFICE_MANAGER') or hasRole('ROLE_ADMIN')")
+    public ResponseEntity<List<DailyAllowanceResponse>> getDailyAllowances() {
+
+        List<DailyAllowanceResponse> dailyAllowances = dailyAllowanceService.getAllDailyAllowances();
+
+        return ResponseEntity.ok().body(dailyAllowances);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<?> getDailyAllowanceByID(@PathVariable Long id) {
+
+        DailyAllowanceResponse dailyAllowance;
+
+        try {
+            dailyAllowance = dailyAllowanceService.findById(id);
+        } catch (DailyAllowanceNotFoundException e) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, e.getMessage(), e);
+        }
+
+        return ResponseEntity.ok().body(dailyAllowance);
+
+    }
+
+    @GetMapping("/search")
+    public ResponseEntity<DailyAllowanceResponse> getDailyAllowanceByRegion(@RequestParam String region) {
+
+        DailyAllowanceResponse dailyAllowance;
+
+        try{
+            dailyAllowance = dailyAllowanceService.findByRegion(region);
```

Comment:
this should not be an exception, it can only return an empty list - meaning no allowances for the region